google-cloud-graalvm-samples/quarkus-pubsub-sample/src/main/java/com/example/PubSubUtils.java

```diff
@@ -0,0 +1,145 @@
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.cloud.pubsub.v1.stub.GrpcSubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStubSettings;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.PullRequest;
+import com.google.pubsub.v1.PullResponse;
+import com.google.pubsub.v1.PushConfig;
+import com.google.pubsub.v1.ReceivedMessage;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ * Simple utility class demonstrating common operations in Google Cloud Pub/Sub.
+ */
+@ApplicationScoped
+public class PubSubUtils {
+  private static final String PROJECT_ID = ServiceOptions.getDefaultProjectId();
+
+  static Topic createTopic(String topicId) throws IOException {
+    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
+      TopicName topicName = TopicName.of(PROJECT_ID, topicId);
+      Topic topic = topicAdminClient.createTopic(topicName);
+      return topic;
+    }
+  }
+
+  static Subscription createSubscription(String subscriptionId, String topicId) throws IOException {
+    try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create()) {
+      TopicName topicName = TopicName.of(PROJECT_ID, topicId);
+      ProjectSubscriptionName subscriptionName =
+          ProjectSubscriptionName.of(PROJECT_ID, subscriptionId);
+      Subscription subscription =
+          subscriptionAdminClient.createSubscription(
+              subscriptionName, topicName, PushConfig.getDefaultInstance(), 10);
+      return subscription;
+    }
+  }
+
+  static void publishMessage(String topicId, String payload) throws Exception {
+    Publisher publisher =
+        Publisher
+            .newBuilder(TopicName.of(PROJECT_ID, topicId))
+            .build();
+
+    ByteString data = ByteString.copyFromUtf8(payload);
+    PubsubMessage pubsubMessage = PubsubMessage.newBuilder().setData(data).build();
+
+    publisher.publish(pubsubMessage);
+
+    ApiFuture<String> messageIdFuture = publisher.publish(pubsubMessage);
+    String messageId = messageIdFuture.get();
+
+    System.out.println("Published message with ID: " + messageId);
+  }
+
+  static String subscribeSync(String subscriptionId) throws IOException {
+    SubscriberStubSettings subscriberStubSettings =
+        SubscriberStubSettings.newBuilder()
+            .setTransportChannelProvider(
+                SubscriberStubSettings.defaultGrpcTransportProviderBuilder()
+                    .setMaxInboundMessageSize(20 * 1024 * 1024) // 20MB (maximum message size).
```

Comment:
That seem excessive, tbh. This is the absolute maximum, and the default value (don't recall, I think it's 4MB) is fine for most purposes.